### PR TITLE
Convert BlockProcessor's child classes to Kotlin

### PR DESCRIPTION
**Pull request by irfano**
This converts classes that inherit from `BlockProcessor` to Kotlin. 

This PR is larger than I initially expected. Let me know if you prefer smaller PRs, and I can divide this into separate PRs for easier review.

I added some comments where special caution is needed during the review.

-----

## To Test:

[test.webm](https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/c92c8ecf-a458-4a2f-a5d8-95372c855706)

To test `BlockProcessor` class, you should follow the steps in the video for the different media types.
1. Edit a post.
2. Add an image.
3. Leave the post before the media upload is done.
4. Wait for uploading to be done.
5. Open the drafted post again.

Repeat these steps for other media blocks as well.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Media upload feature

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Updated current tests

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] ~WordPress.com sites and self-hosted Jetpack sites.~
- [ ] ~Portrait and landscape orientations.~
- [ ] ~Light and dark modes.~
- [ ] ~Fonts: Larger, smaller and bold text.~
- [ ] ~High contrast.~
- [ ] ~Talkback.~
- [ ] ~Languages with large words or with letters/accents not frequently used in English.~
- [ ] ~Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)~
- [ ] ~Large and small screen sizes. (Tablet and smaller phones)~
- [ ] ~Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.kt [hunk 1]
```diff
@@ -72,37 +72,37 @@ abstract class BlockProcessor internal constructor(@JvmField var localId: String
...
+                .append(blockName)
+                .append(" ")
+                .append(jsonAttributes) // json parser output
+                .append(" -->\n")
+                .append(blockContentDocument?.body()?.html()) // HTML parser output
+                .append(closingComment)
+                .toString()
+        }
+
+        else -> block // leave block unchanged
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.kt:105]: Please review this `processBlock()` function more carefully to ensure that I didn't change any behavior.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                        ids[index] = JsonPrimitive(remoteId.toInt(RADIX))
+                    } catch (e: NumberFormatException) {
+                        AppLog.e(AppLog.T.MEDIA, e.message)
+                    }
+                    return true
+                }
+            }
+        }
+        return false
+    }
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/GalleryBlockProcessor.kt:81]: Please review this `processBlockJsonAttributes()` function more carefully.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.kt (hunk 1 above)
Comment (notandyvee) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.kt:105]: Won't lie. This was hard to follow 😅 . The function appears same as before though 👍🏻

**Review by notandyvee — APPROVED** [review]
Reviewed and it LGTM. Tested all the block processors and no issues. Good work 👍🏻
